gapis/api/vulkan/framegraph.go

```diff
@@ -209,6 +302,28 @@ func (helpers *framegraphInfoHelpers) processSubCommand(ctx context.Context, dep
 	if helpers.rpInfo != nil {
 		nodeID := dependencyGraph.GetCmdNodeID(api.CmdID(subCmdIdx[0]), subCmdIdx[1:])
 		helpers.rpInfo.nodes = append(helpers.rpInfo.nodes, nodeID)
+
+		for _, memAccess := range dependencyGraph.GetNodeAccesses(nodeID).MemoryAccesses {
```

Comment:
You may want to cache this unless you are modifying it in the loop. I am not sure, double check this, but go may try to call dependencyGraph.GetNodeAccesses(nodeID) in every cycle in the loop.

Reply:
Go will evaluate it only once, the Go spec says (https://golang.org/ref/spec#For_statements , section For statements with range clause):

> The range expression x is evaluated once before beginning the loop, with one exception: if at most one iteration variable is present and len(x) is constant, the range expression is not evaluated.